### PR TITLE
fix: added github html url to task request

### PR DESCRIPTION
**Pull request by sahsisunny**
## Issue Ticket Number:- 
-  Real-Dev-Squad/website-status#1050

## Description: 
In this pull request, I have addressed the issue described in Real-Dev-Squad/website-status#1050. The problem was related to the absence of GitHub issue ticket links for tasks created via task requests. 

Is Under Feature Flag 
- [x] Yes
- [ ] No

Database changes
- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)
- [ ] Yes
- [x] No

Is Development Tested?

- [x] Yes
- [ ] No

Tested in staging?

- [x] Yes
- [ ] No

### Add relevant Screenshot below ( e.g test coverage etc. )
File                              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------------------------------|---------|----------|---------|---------|-------------------          
 website-backend/models           |   54.74 |     45.8 |   52.86 |   55.11 |                   
  taskRequests.js                 |   86.31 |    83.87 |   86.36 |   87.35 | ...15-351,382,529 

## Demo

https://github.com/Real-Dev-Squad/website-backend/assets/70854507/10367c42-39aa-407d-93e8-2ab7d1565bac

# Related
PR - https://github.com/Real-Dev-Squad/website-status/pull/1051

### CODE REVIEW

**Review by Ajeyakrishna-k — DISMISSED** [review]
Please don't transform the url which is in task requests. Task requests are not tasks, GitHub url should be present on tasks. So change/add the url in tasks document created from task requests. 

- This change will BREAK multiple task requests flow. 
- BREAK the GitHub issue render in task request details page

File: middlewares/validators/task-requests.js [hunk 1]
```diff
@@ -24,6 +27,7 @@ const postTaskRequests = async (req, res, next) => {
     .strict()
     .keys({
       externalIssueUrl: joi.string().regex(githubIssuerUrlPattern).required(),
+      externalIssueHtmlUrl: joi.string().regex(githubIssueHtmlUrlPattern).required(),
```
Comment (satyam73) [inline middlewares/validators/task-requests.js:30]: required here and optional above, not getting it?

File: middlewares/validators/task-requests.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
+import { GITHUB_URL } from "../../constants/urls";
 const joi = require("joi");
 const { RQLQueryParser } = require("../../utils/RQLParser");
 const githubOrg = config.get("githubApi.org");
 const githubBaseUrl = config.get("githubApi.baseUrl");
 const githubIssuerUrlPattern = new RegExp(`^${githubBaseUrl}/repos/${githubOrg}/.+/issues/\\d+$`);
+const githubIssueHtmlUrlPattern = new RegExp(`^${GITHUB_URL}/${githubOrg}/.+/issues/\\d+$`);
```
Comment (satyam73) [inline middlewares/validators/task-requests.js:7]: Please add what type of pattern are you checking against with a example in description

Comment (sahsisunny) [reply middlewares/validators/task-requests.js]: added

File: middlewares/validators/task-requests.js (hunk 1 above)
Comment (sahsisunny) [reply middlewares/validators/task-requests.js:30]: fixed

**Review by satyam73 — DISMISSED** [review]
LGTM

File: constants/urls.ts
[the remark this answers is not in the document]
Comment (sahsisunny) [reply constants/urls.ts:1]: No

File: test/unit/models/task-requests.test.js
[the remark this answers is not in the document]
Comment (sahsisunny) [reply test/unit/models/task-requests.test.js:1]: For what?

**Review by vinit717 — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**